templates.rst

```diff
@@ -1631,9 +1633,10 @@ this command to confirm that your new filter was successfully registered:
 Creating Lazy-Loaded Twig Extensions
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Including the code of the custom filters/functions in the Twig extension class
-is the simplest way to create extensions. However, Twig must initialize all
-extensions before rendering any template, even if the template doesn't use an
+When using attributes to extend Twig, the services are initialized only when
+the functions or filters are used to render the template. But in case you use the
+classic approach by extending the ``AbstractExtension`` class, Twig initialize all extensions before
+rendering rendering any template, even if the template doesn't use an
 extension.
```

Comment:
```suggestion
When using attributes to extend Twig, the services are initialized only when
the functions or filters are used to render the template. But in case you use the
classic approach by extending the ``AbstractExtension`` class, Twig initializes all the extensions before
rendering any template, even if the extension is not used in the template.
```